pilot/pkg/autoregistration/controller.go

```diff
@@ -248,7 +248,15 @@ func (c *Controller) registerWorkload(entryName string, proxy *model.Proxy, conT
 		log.Infof("updated auto-registered WorkloadEntry %s/%s", proxy.Metadata.Namespace, entryName)
 		return nil
 	}
-
+	// at this point, we list all WorkloadEntries with the same IP and network, and clean them up (they are stale).
+	entries := c.store.List(gvk.WorkloadEntry, proxy.Metadata.Namespace)
```

Comment:
We cannot make sure the store must returns all the WLE that has been created, because it it cache here

Reply:
Does store ensure that we can read our own writes immediately?

i.e
```
Create(x)
Read(x) // Must always be present
```

Reply:
Actually not sure it matters since we may multiple instances

Reply:
I think we would probably need a controller that watches WE and on each update queries for all others with the same ip/network (probably an index is useful here) and removes them. Anything relying on connection as the trigger for cleanup is going to be racy

Reply:
so rely on CRD update as opposed to WE connect? i'll add an onUpdate function that does this